### PR TITLE
[T5A1][W11-B3]Fan Wegiuang

### PR DESCRIPTION
**Pull request by fanwgwg**
Ready for review
The first two commits should be ignored, because I forgot to checkour master branch before creating the branch for next activity, so I reverted the commit for this activity.

### CODE REVIEW

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -73,13 +75,47 @@ Use case ends.
...
+6. AddressBook shows the new list of tags
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> Use case ends
+
+3a. The given index is invalid
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md]: There isn't any index mentioned in that step, may be `the given tag name to be edited is invalid`

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -73,13 +75,47 @@ Use case ends.
...
+
+> 3a1. AddressBook shows an error message <br>
+  Use case resumes at step 2
+
+3b. The command user given is invalid
+
+> 3b1. AddressBook shows and error message <br>
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md]: Just a typo. :)

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -73,13 +75,47 @@ Use case ends.
...
 
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java 8 or higher installed.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should contain a user guide explaining how to use.
+6. Should contain a developer guide to explain how this product is designed and implemented.
+7. Should work on a desktop without Internet connection.
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md]: It may be included in the extension part of User stories.

File: doc/DeveloperGuide.md [hunk 4]
```diff
@@ -73,13 +75,47 @@ Use case ends.
...
+8. Should not use relational databases.
+9. Should store date locally in a human editable textfile.
```
Comment (jeffryhartanto) [inline doc/DeveloperGuide.md]: Point 8 and 9 can be combined since both of them concern about the storage.

**Comment by jeffryhartanto** [comment]
Some comments added. Please close the PR after reading comments.

**Comment by fanwgwg** [comment]
Comments reviewed. Thank youvery much!

**Pull request closed without merging**